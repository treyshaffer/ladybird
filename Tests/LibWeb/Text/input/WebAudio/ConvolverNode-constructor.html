<!DOCTYPE html>
<script src="../include.js"></script>
<script>
test(() => {
    const context = new AudioContext();
    
    // Test ConvolverNode exists
    println("ConvolverNode constructor exists: " + (typeof ConvolverNode !== 'undefined'));
    
    // Test constructor
    let node;
    try {
        node = new ConvolverNode(context);
        println("ConvolverNode created successfully");
    } catch (e) {
        println("Failed to create ConvolverNode: " + e.message);
        return;
    }
    
    // Test properties
    println("numberOfInputs: " + node.numberOfInputs);
    println("numberOfOutputs: " + node.numberOfOutputs);
    println("channelCount: " + node.channelCount);
    println("channelCountMode: " + node.channelCountMode);
    println("channelInterpretation: " + node.channelInterpretation);
    println("normalize: " + node.normalize);
    println("buffer: " + node.buffer);
    
    // Test with options
    try {
        const node2 = new ConvolverNode(context, { disableNormalization: true });
        println("ConvolverNode with disableNormalization created");
        println("normalize (with disableNormalization=true): " + node2.normalize);
    } catch (e) {
        println("Failed with options: " + e.message);
    }
    
    // Test factory method
    try {
        const node3 = context.createConvolver();
        println("context.createConvolver() successful");
    } catch (e) {
        println("Factory method failed: " + e.message);
    }
});
</script>