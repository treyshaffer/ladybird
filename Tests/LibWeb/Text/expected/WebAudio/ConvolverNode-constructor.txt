ConvolverNode constructor exists: true
ConvolverNode created successfully
numberOfInputs: 1
numberOfOutputs: 1
channelCount: 2
channelCountMode: clamped-max
channelInterpretation: speakers
normalize: true
buffer: null
ConvolverNode with disableNormalization created
normalize (with disableNormalization=true): false
context.createConvolver() successful